libgadget/blackhole.c

```diff
@@ -84,7 +84,8 @@ typedef struct {
 } TreeWalkNgbIterBHFeedback;
 
 struct BHPriv {
-    /*Temporary array to store the IDs of the swallowing black hole for gas*/
+    /* Temporary array to store the IDs of the swallowing black hole for gas.
+     * We store ID + 1 so that SwallowID == 0 can correspond to the unswallowed case. */
```

Comment:
Might be cleaner using 0xffff ffff ffff ffff for un-swallowed.

Reply:
Then I can't use < cleanly and I need to worry about integer overflow. This is a temporary only used in this routine so a constant offset is not a big deal.